cmd/config_merger/main.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"io/ioutil"
+	"time"
+
+	"github.com/GoogleCloudPlatform/testgrid/pkg/merger"
+	"github.com/GoogleCloudPlatform/testgrid/util/gcs"
+	"github.com/sirupsen/logrus"
+)
+
+type options struct {
+	listPath     string
+	creds        string
+	confirm      bool
+	wait         time.Duration
+	skipValidate bool
+}
+
+func (o *options) validate() error {
+	if o.listPath == "" {
+		return errors.New("--config-list of configurations to merge required")
+	}
+	if !o.confirm {
+		logrus.Info("--confirm=false (DRY-RUN): will not write to gcs")
+	}
+	if o.skipValidate {
+		logrus.Info("--allow-invalid-configs: result may not validate either")
+	}
+	return nil
+}
+
+func gatherOptions() options {
+	var o options
+	flag.StringVar(&o.listPath, "config-list", "", "List of configurations to merge")
+	flag.StringVar(&o.creds, "gcp-service-account", "", "/path/to/gcp/creds (use local creds if empty)")
+	flag.BoolVar(&o.confirm, "confirm", false, "Upload data if set")
+	flag.DurationVar(&o.wait, "wait", 0, "Ensure at least this much time ahs passed since the last loop. (Run only once if zero)")
+	flag.BoolVar(&o.skipValidate, "allow-invalid-configs", false, "Allows merging of configs that don't validate. Usually skips invalid configs")
+	flag.Parse()
+	return o
+}
+
+func main() {
+	opt := gatherOptions()
+	err := opt.validate()
+	if err != nil {
+		logrus.WithError(err).Fatal("Invalid flags")
+	}
+	file, err := ioutil.ReadFile(opt.listPath)
+	if err != nil {
+		logrus.WithError(err).Fatalf("Can't read file at %s", opt.listPath)
```

Comment:
logrus.WithField("--config-list", opt.ListPath).WithError(err).Fata("Cannot find config")

Reply:
Done here and with similar logging calls